### PR TITLE
erofs-utils: 1.2.1 -> 1.3

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

New features in 1.3.

(Someone still needs to test replacing squashfs with erofs in the NixOS images)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137172` at 46da1e10 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>erofs-utils</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

autoreconfHook is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/os-specific/linux/erofs-utils/default.nix:15:3:
```
   |
15 |   buildInputs = [ autoreconfHook pkg-config fuse libuuid lz4 ];
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

pkg-config is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/os-specific/linux/erofs-utils/default.nix:15:3:
```
   |
15 |   buildInputs = [ autoreconfHook pkg-config fuse libuuid lz4 ];
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/erofs-utils/default.nix:21:5:
```
   |
21 |     license = with licenses; [ gpl2 ];
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/os-specific/linux/erofs-utils/default.nix [hunk 1]
```diff
@@ -2,23 +2,24 @@
...
-  buildInputs = [ autoreconfHook pkg-config fuse libuuid lz4 ];
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [ fuse libuuid lz4 ];
 
   configureFlags = [ "--enable-fuse" ];
 
   meta = with lib; {
     description = "Userspace utilities for linux-erofs file system";
-    license = with licenses; [ gpl2 ];
+    license = with licenses; [ gpl2Only ];
```
Comment (symphorien) [inline pkgs/os-specific/linux/erofs-utils/default.nix:22]: https://git.kernel.org/pub/scm/linux/kernel/git/xiang/erofs-utils.git/tree/mkfs/main.c#n1 says gpl2Plus

File: pkgs/os-specific/linux/erofs-utils/default.nix [hunk 2]
```diff
@@ -2,23 +2,23 @@
...
 
-  buildInputs = [ autoreconfHook pkg-config fuse libuuid lz4 ];
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [ fuse libuuid lz4 ];
 
   configureFlags = [ "--enable-fuse" ];
 
   meta = with lib; {
     description = "Userspace utilities for linux-erofs file system";
-    license = with licenses; [ gpl2 ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/erofs-utils/default.nix:21]: Why is this removed?

File: pkgs/os-specific/linux/erofs-utils/default.nix [hunk 3]
```diff
@@ -2,23 +2,23 @@
 
 stdenv.mkDerivation rec {
   pname = "erofs-utils";
-  version = "1.2.1";
+  version = "1.3";
   outputs = [ "out" "man" ];
 
   src = fetchgit {
     url =
       "https://git.kernel.org/pub/scm/linux/kernel/git/xiang/erofs-utils.git";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/erofs-utils/default.nix:10]:
```suggestion
    url = "https://git.kernel.org/pub/scm/linux/kernel/git/xiang/erofs-utils.git";
```
